### PR TITLE
feat: Add hyperlink for assessments in Dates Tab

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description:

- Update CourseDateBlock Model
- Add Underline text decorations for clickable text
- Navigate to Component Screen by tapping hyperlinks
- Maintain stack of navigation

### CODE REVIEW

**Comment by miankhalid** [comment]
@volodymyr-chekyrta any update on the review of this PR?

**Review by k1rill — APPROVED** [review]

File: core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt [hunk 1]
```diff
@@ -8,11 +9,16 @@ data class CourseDateBlock(
...
+    @SerializedName("learner_has_access")
     val learnerHasAccess: Boolean = false,
     val complete: Boolean = false,
     val date: Date?,
+    @SerializedName("date_type")
     val dateType: DateType = DateType.NONE,
+    @SerializedName("assignment_type")
+    val assignmentType: String = "",
     var dateBlockBadge: CourseDatesBadge = CourseDatesBadge.BLANK,
 ) {
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt:23]: Could you please clarify why it's necessary to include SerializedName annotations in the domain model?
Usually, data models are used for serialization but not domain ones.

**Comment by volodymyr-chekyrta** [comment]
> @volodymyr-chekyrta any update on the review of this PR?

From what I understand, both @farhan-arshad-dev and @HamzaIsrar12 are yet to review the PR. 
As previously dicscussed, it would be most efficient if my review will be the last one.

File: core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt (hunk 1 above)
Comment (omerhabib26) [reply core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt:23]: Removed

**Comment by omerhabib26** [comment]
@HamzaIsrar12 is currently reviewing it

File: core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt (hunk 1 above)
Comment (volodymyr-chekyrta) [reply core/src/main/java/org/openedx/core/domain/model/CourseDateBlock.kt]: Thank you!

File: course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt [hunk 2]
```diff
@@ -143,23 +145,40 @@ class CourseSectionFragment : Fragment() {
...
                 )
+
+                LaunchedEffect(rememberSaveable { true }) {
+                    val descendantId = requireArguments().getString(ARG_DESCENDANT_ID, null)
+                    if(descendantId.isNotEmpty()) {
+                        router.navigateToCourseContainer(
+                            requireActivity().supportFragmentManager,
+                            descendantId,
+                            courseId = viewModel.courseId,
+                            courseName = "",
```
Comment (HamzaIsrar12) [inline course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt:156]: Is it okay to pass an empty string? 🤔

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]

File: course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt (hunk 2 above)
Comment (omerhabib26) [reply course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt:156]: we no longer need `courseName` in descendant views, so I have updated the code accordingly

File: app/src/main/java/org/openedx/app/AppRouter.kt [hunk 3]
```diff
@@ -116,36 +116,35 @@ class AppRouter : AuthRouter, DiscoveryRouter, DashboardRouter, CourseRouter, Di
         courseId: String,
         blockId: String,
         mode: CourseViewMode,
+        descendantId : String?
```
Comment (HamzaIsrar12) [inline app/src/main/java/org/openedx/app/AppRouter.kt:119]: Needs autoformat.

File: course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt [hunk 4]
```diff
@@ -143,23 +144,39 @@ class CourseSectionFragment : Fragment() {
                         }
                     }
                 )
+
+                LaunchedEffect(rememberSaveable { true }) {
+                    val descendantId = requireArguments().getString(ARG_DESCENDANT_ID, "")
+                    if(descendantId.isNotEmpty()) {
```
Comment (HamzaIsrar12) [inline course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt:150]: Requires formatting and a new line at the end of the file.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Just some minor nits.

File: course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt [hunk 5]
```diff
@@ -143,23 +145,40 @@ class CourseSectionFragment : Fragment() {
                         }
                     }
                 )
+
+                LaunchedEffect(rememberSaveable { true }) {
+                    val descendantId = requireArguments().getString(ARG_DESCENDANT_ID, null)
+                    if(descendantId.isNotEmpty()) {
```
[the remark this answers is not in the document]
Comment (omerhabib26) [reply course/src/main/java/org/openedx/course/presentation/section/CourseSectionFragment.kt]: default value update with  empty string `""`

**Review by HamzaIsrar12 — APPROVED** [review]
LGTM 🏎️

**Review by k1rill — APPROVED** [review]

**Pull request merged**